components/hab/mac/README.md

```diff
@@ -1,26 +1,74 @@
 
-* Xcode CLI tools
-* Homebrew
-* `coreutils`, `gnu-tar`, and `wget` Homebrew packages
-* Rust
-* Cargo nightly
+## Prerequisites
 
-A final prerequisite of a `hab` binary itself is required by `hab-plan-build.sh` to sign the resulting Habitat artifact. The `mac-build.sh` will install the latest release from Bintray if `/bin/hab` cannot be found locally, otherwise the `hab` on your `PATH` will be used. As we are using the build program, a secret origin key is also required in the Mac's key cache under `/hab/cache/keys`.
-
-## Usage
+### Install XCode Command Line Developer Tools
+This installs basic developer tooling needed to compile and build software on a Mac.
 
 ```sh
-cd components/hab/mac
-sudo ./mac-build.sh
+xcode-select --install
 ```
 
-Assuming success, this will produce a local `./results` directory with the artifact.
+### Install Omnibus Bootstrap Toolchain package
+
+Since there is not yet a complete Habitat build toolchain available for MacOS, we provide the minimal set of binaries and static libraries needed to compile a `hab` binary using Chef's Omnibus tooling platform. This effectively takes the place of the packages we would add to a `pkg_build_deps` entry in a Habitat plan file.
 
-Alternatively, as `mac-build.sh` is a wrapper around the build program, it can be just as easily invoked from the root of the source tree with:
+TODO: Where can this package be retrieved from?
 
 ```sh
-sudo ./components/hab/mac/mac-build.sh components/hab/mac
+sudo installer \
+     -pkg <PATH_TO_BOOTSTRAP_TOOLCHAIN_PACKAGE> \
+     -target /
+```
+
+TODO: Can we rename `hab-bundle`?
+
+This will install the toolchain in `/opt/hab-bundle`; this is where the build program is expecting to find binaries and libraries it needs.
+
+### Install Homebrew
+Follow the instructions at https://brew.sh.
+
+### Install Rust toolchain
+Follow the instructions at https://rustup.rs/
+
+### Install Homebrew prerequisites
+Though most of the toolchain needed to build a `hab` binary exist in the Omnibus bootstrap toolchain, a few are not yet available there. Until they are migrated in, we need to use Homebrew to get them.
+
+``` sh
+brew bundle install --verbose --file=<HABITAT_REPO>/.buildkite/Brewfile
+```
+
+This step is temporary, until we have a fully-contained toolchain in an Omnibus package
+
+### Install `hab`
+This is currently handled with the `brew bundle` command above, but for completeness:
+
+``` sh
+brew tap habitat-sh/habitat
+brew install hab
+```
+
+### Install Builder Origin Keys
+
+To build a Habitat package, you must have the secret `core` origin key for signing that package. Obtaining the secret key requires a personal authentication token from Builder.
+
+``` sh
+sudo hab origin key download core
+sudo hab origin key download --secret --auth="${HAB_AUTH_TOKEN}" core
```

Comment:
While this won't bite us here, it might be more consistent in the future to use `sudo -E` as it will pass currently exported environment variables to the command to be run. I see that form used a bit later in the instructions which makes sense.

Reply:
We should also try to reduce the use of `sudo -E` as much as possible, in case that hasn't be stated. I'm a big fan of explicit passing vs "passing the environment through"

Reply:
After thinking about it a bit, I think I'm also in favor of reducing `sudo -E` in favor of being super-explicit. Any `sudo -E`s that I added are probably bits of early "just make it work" code that I didn't fully clean up.